src/HttpHelper/HttpClientHelper/HttpClient.php

```diff
@@ -7,80 +7,111 @@
+    /**
+     * @throws GuzzleHttp\Exception\GuzzleException
+     */
+    public function patch(string $sUrl, array $aRequestOptions, string $sAction, LogContext $logContext, array $aLogMetadata = []): ResponseInterface
     {
-        $aLogMetadata['url'] = $sUrl;
-        $aLogMetadata['httpCode'] = $response->getHttpCode();
-        $aLogMetadata['response'] = $response->getResponse();
-        $request && $aLogMetadata['request'] = $request->getBody();
-
-        if (!$response->isSuccess()) {
-            Logger::error("$sAction has failed: {$response->getErrorMessage()}", $logContext, $aLogMetadata);
-        } else {
-            Logger::debug("$sAction succeed", $logContext, $aLogMetadata);
-        }
+        return $this->request(Request::METHOD_PATCH, $sUrl, $aRequestOptions, $sAction, $logContext, $aLogMetadata);
     }
 
-    private function request(string $sMethod, string $sUrl, array $aRequestOptions, $fRequestTimeout = null): HttpClientResponse
+    private function request(
+        string $sMethod,
+        string $sUrl,
+        array $aRequestOptions,
+        string $sAction,
+        LogContext $logContext,
+        array $aLogMetadata = []
+    ): ResponseInterface
     {
-        $bError = true;
         try {
-            $aRequestOptions[RequestOptions::TIMEOUT] = $fRequestTimeout ? : self::DEFAULT_REQUEST_TIMEOUT;
+            $aRequestOptions[RequestOptions::TIMEOUT] = $aRequestOptions[RequestOptions::TIMEOUT] ?? self::DEFAULT_REQUEST_TIMEOUT;
             $response = $this->httpClient->request($sMethod, $sUrl, $aRequestOptions);
+
             $sResponseBody = $response->getBody()->getContents();
-            $iHttpCode = $response->getStatusCode();
-            $bError = false;
-        } catch (ConnectException $e) {
-            $sResponseBody = "ConnectException: {$e->getMessage()}";
-            $iHttpCode = 0;
+            $response->getBody()->rewind();
+            Logger::debug("$sAction : succeed requesting $sMethod $sUrl - {$response->getStatusCode()}", $logContext, \array_merge($aLogMetadata, [
+                'response' => $sResponseBody,
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
         } catch (GuzzleException $e) {
-            if (\method_exists($e, 'hasResponse') && $e->hasResponse()) {
-                $response = $e->getResponse();
-                $sResponseBody = $response->getBody()->getContents();
-                $iHttpCode = $response->getStatusCode();
-            } else {
-                $sResponseBody = "GuzzleException: {$e->getMessage()}";
-                $iHttpCode = 0;
-            }
+            Logger::debug("$sAction : error while requesting $sMethod $sUrl - {$e->getCode()}", $logContext, \array_merge($aLogMetadata, [
+                'response' => $e->getMessage(),
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
+
+            throw $e;
```

Comment:
In kali in order to have the exception message we just do:
`$exception->getMessage()`

But this is not enough, because the message that we get is something like this:
```
Client error: `GET https://api.floapay.io/cdp-preprod/api/v3/fundings/20329786` resulted in a `401 Unauthorized` response:
{"code":401,"codeString":"AUTH_TOKEN_EXPIRED","message":"The client auth. token is expired. Please renew it."}
```
Which is not great because our message is always truncated. For long errors we can get all errors.

I found yesterday that catching BadResponseException we can always access the original response. This alwready what you did in your previous code by checking if method exists and then try to access it.

```
 $response = $e->getResponse();
 $sResponseBody = $response->getBody()->getContents();
```
Yesterday when I tried to get the original response from the guzzleResponse I was never able to get the original response. I got it only by catching BadResponseException also with something like this:

`catch (GuzzleException | BadResponseException $e) {`

Thanks to that I was able to get the original response. Can we have something similar ? Is really important for kali to have the orignal response so we can access all errors.  By throwing $e we will be able to access easily the original response.

I test it on kali so on guzzle 7 I don't know if it's the same than guzzle 6

Reply:
@nowfel-t  @leadinpier You see, that was purpose of the previous version of `HttpClient`, to make Your life easier and to avoid playing with Guzzle exceptions. `HttpClient` was doing it for You and was returning the full response body and the http code (I believe two things You are really interested in when doing request).
But You wanted to catch exceptions by Yourself so now You need to handle them.

`BadResponseException` is `GuzzleException` so it will be re-thrown by `HttpClient`, no need to change anything.

Reply:
I've made a little improvement. `HttpClient` throws now custom exception. What You need is just catch `HttpClientException` and You can get full response `$e->getMessage()` and the http code `$e->getCode()`

Reply:
Ummmm I'm not really okay with that. I mean it makes life easier for logs but not for exception. But for the try catch no. I means yes in a certain way, but when I need control, it makes things complicated.
_But You wanted to catch exceptions by Yourself so now You need to handle them._ Yes and it's exatly what I want but I need the right exception for it that's all.

Reply:
The only issue I see is that devs can't handle these exceptions properly. That's why we want to do it in the bundle.
Tbh I did not see the code in our stack which catches different guzzle exceptions and do something depends on the exception type. But I saw plenty of code where guzzle exceptions are handled badly.

Reply:
ok

Reply:
Completely I'm not taking it personally. I see Your point, but we see it's not working. You will be able to catch Your own exception but it will be `HttpClient` exception ;)

Reply:
Sure, it's not final version. It can be enriched at any moment, depends on the needs.